src/Client.ts

```diff
@@ -263,27 +258,27 @@ export default class Client {
     if (
       bundle &&
       bundle instanceof SignedPublicKeyBundle &&
-      this.keys.getPublicKeyBundle().equals(bundle)
+      this.signedPublicKeyBundle.equals(bundle)
     ) {
       return
     }
     // TEMPORARY: publish V1 contact to make sure there is one in the topic
     // in order to preserve compatibility with pre-v7 clients.
     // Remove when pre-v7 clients are deprecated
-    this.publishUserContact(true)
+    await this.publishUserContact(true)
```

Comment:
I caught this bug along the way. We should await the publishing of the contacts to ensure that no races can change the ordering of the bundles. If the legacy bundle is the newest one, other clients may attempt to create V1 conversations instead of V2.

Reply:
nice find! Do we normally get warnings emitted for unchecked Promise or something similar? Wondering if there are any more bugs like this lurking around.

Reply:
This does seem like something we should be able to catch with a linter. I'll look into that.